test/e2e/framework/networking_utils.go

```diff
@@ -76,6 +77,16 @@ const (
 
 var netexecImageName = imageutils.GetE2EImage(imageutils.Agnhost)
 
+// TranslateIPv4ToIPv6 maps an IPv4 address into a valid IPv6 address
+// adding the well known prefix "0::ffff:" https://tools.ietf.org/html/rfc2765
+// if the ip is IPv4 and the cluster IPFamily is IPv6, otherwise returns the same ip
+func TranslateIPv4ToIPv6(ip string) string {
+	if TestContext.IPFamily == "ipv6" && !k8utilnet.IsIPv6String(ip) {
+		ip = "0::ffff:" + ip
```

Comment:
Why not just ` ::ffff:?

Reply:
I was doing to use it, but wanted to keep it consistent with the rfc https://tools.ietf.org/html/rfc2765
> 2.1.  Addresses

   In addition to the forms of addresses defined in [ADDR-ARCH] this
   document also introduces the new form of IPv4-translated address.
   This is needed to avoid using IPv4-compatible addresses outside the
   intended use of automatic tunneling.  Thus the address forms are:

         IPv4-mapped:
                 An address of the form 0::ffff:a.b.c.d which refers
                 to a node that is not IPv6-capable.

Reply:
Good to know, thanks!